### PR TITLE
Fix multi select sync for Hubspot

### PR DESCRIPTION
**Pull request by kuzmany**
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Fix for both way sync multi select field for Hubspot integration

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1 Create a custom multiple field in Mautic
2 Create a custom multiple field in Hubspot with the same exact choices as in Mautic
3 Create a contact in Mautic with at least 2 choices in the multiple field and push it in Hubspot with a form
4 See that in Hubspot the multiple field is empty
5 Create a contact in Hubspot and synchronize it with cron to Mautic.
6 See that in Mautic the multiple field is empty

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Should works properly

#### List deprecations along with the new alternative:
1. 
2. 

#### List backwards compatibility breaks:
1. 
2.

### CODE REVIEW

**Comment by Dcoutelle** [comment]
@kuzmany , my bad, I did an error when I applied your patch... I just redid it, Florent is currently testing it.

File: plugins/MauticCrmBundle/Integration/CrmAbstractIntegration.php [hunk 1]
```diff
@@ -432,6 +432,10 @@ public function getMauticLead($data, $persist = true, $socialCache = null, $iden
                 return;
             }
 
+            if (isset($fieldsToUpdateInMautic[0])) {
```
Comment (Dcoutelle) [inline plugins/MauticCrmBundle/Integration/CrmAbstractIntegration.php]: @kuzmany we are 100% sure there is no problem with other integration ? Maybe we should check if we are in hubspot context ?

**Comment by kuzmany** [comment]
@Dcoutelle I reworked it directly in Hubspot integration
@florentpetitjean please re-test again deeply

**Comment by npracht** [comment]
Hi @afonseca08 and @ brendansmallco i saw you use this integration. Could you please test https://github.com/mautic/mautic/pull/7887? Your help would be highly appreciated to make it merged faster.
If you need any help or hints to know how to test, please read this guide: [Become a community contributor by easily testing mautic](https://www.mautic.org/blog/community/become-a-community-contributor-by-easily-testing-mautic/).

Thanks a lot for your help!

**Comment by brendansmallco** [comment]
hey @npracht happy to test, but I cant provide my customer's actual hubspot API creds. Is there a test instance? 

p.s. ive replaced my own sandbox instance with the new test HubspotIntegration.php and MySQL is still throwing errors - Hubspot is providing a millisecond timestamp - SQL isnt quite liking it - Logs:

[2019-10-07 09:39:53] mautic.WARNING: An exception occurred while executing 'UPDATE ma_leads SET persona = ?, persona_orientation = ?, product_segment = ?, lead_create_date = ?, hubspot_source = ? WHERE id = ?' with params ["persona_8", "Data Orientated (Revenue)", "Spray", "1568857151139", "OFFLINE", 32]:  SQLSTATE[22007]: Invalid datetime format: 1292 Incorrect datetime value: '1568857151139' for column 'lead_create_date' at row 1 [] []
[2019-10-07 09:39:53] mautic.WARNING: An exception occurred while executing 'UPDATE ma_leads SET lead_create_date = ?, hubspot_source = ? WHERE id = ?' with params ["1569365521385", "OFFLINE", 37]:  SQLSTATE[22007]: Invalid datetime format: 1292 Incorrect datetime value: '1569365521385' for column 'lead_create_date' at row 1 [] []

**Comment by npracht** [comment]
This issue is not fixing date/time field. Please unmap this field for the test and just try to sync. multiselect field (this is what is supposed to be fixed by this PR).

**Comment by npracht** [comment]
Hi @kuzmany  can you check if your PR is mergeable in https://github.com/mautic/mautic/tree/3.x and manage conflicts if needed?
Then we give a chance to the PR to be merged in 3.0.0. Thanks !

**Comment by npracht** [comment]
Hi @kuzmany  can you check if your PR is mergeable in https://github.com/mautic/mautic/tree/3.x and manage conflicts if needed?
Then we give a chance to the PR to be merged in 3.0.0. Thanks !

**Pull request merged**